src/main/java/edu/harvard/iq/dataverse/DatasetServiceBean.java

```diff
@@ -1042,4 +1061,492 @@ public void deleteHarvestedDataset(Dataset dataset, DataverseRequest request, Lo
+                    }
+
+                    // calculateMissingMetadataFields: checksum, mimetype
+                    JsonObject newfilesJsonObject = calculateMissingMetadataFields(inputList, globusLogger);
+                    JsonArray newfilesJsonArray = newfilesJsonObject.getJsonArray("files");
+
+                    JsonArrayBuilder jsonDataSecondAPI = Json.createArrayBuilder();
+
+                    for (JsonObject fileJsonObject : filesJsonArray.getValuesAs(JsonObject.class)) {
+
+                        countAll++;
+                        String storageIdentifier = fileJsonObject.getString("storageIdentifier");
+                        String fileName = fileJsonObject.getString("fileName");
+                        String directoryLabel = fileJsonObject.getString("directoryLabel");
+                        String[] bits = storageIdentifier.split(":");
+                        String fileId = bits[bits.length - 1];
+
+                        List<JsonObject> newfileJsonObject = IntStream.range(0, newfilesJsonArray.size())
+                                .mapToObj(index -> ((JsonObject) newfilesJsonArray.get(index)).getJsonObject(fileId))
+                                .filter(Objects::nonNull).collect(Collectors.toList());
+
+                        if (newfileJsonObject != null) {
+                            if ( !newfileJsonObject.get(0).getString("hash").equalsIgnoreCase("null")) {
+                                JsonPatch path = Json.createPatchBuilder().add("/md5Hash", newfileJsonObject.get(0).getString("hash")).build();
+                                fileJsonObject = path.apply(fileJsonObject);
+                                path = Json.createPatchBuilder().add("/mimeType", newfileJsonObject.get(0).getString("mime")).build();
+                                fileJsonObject = path.apply(fileJsonObject);
+                                jsonDataSecondAPI.add(stringToJsonObjectBuilder(fileJsonObject.toString()));
+                                countSuccess++;
+                            } else {
+                                globusLogger.info(fileName + " will be skipped from adding to dataset by second API due to missing values ");
+                                countError++;
+                            }
+                        } else {
+                            globusLogger.info(fileName + " will be skipped from adding to dataset by second API due to missing values ");
+                            countError++;
+                        }
+                    }
+
+                    String newjsonData = jsonDataSecondAPI.build().toString();
+
+                    globusLogger.info("Successfully generated new JsonData for Second API call");
+
+
+                    String command = "curl -H \"X-Dataverse-key:" + token.getTokenString() + "\" -X POST " + httpRequestUrl + "/api/datasets/:persistentId/addFiles?persistentId=doi:" + datasetIdentifier + " -F jsonData='" + newjsonData + "'";
```

Comment:
Could/should probably just call the Datasets.addFilesToDataset method directly (or refactor to move the relevant parts to AddFileReplaceHelper?)